server/plugin/virtual_agent_test.go

```diff
@@ -452,3 +455,120 @@ func Test_CreateMessageAttachment(t *testing.T) {
+	p := Plugin{}
+	for _, test := range []struct {
+		description   string
+		expectedError error
+		setupPlugin   func(p *Plugin, api *plugintest.API)
+		setupStore    func(s *mock_plugin.MockStore)
+	}{
+		{
+			description: "carousel is handled successfully with no error",
+			setupPlugin: func(p *Plugin, api *plugintest.API) {
+				monkey.PatchInstanceMethod(reflect.TypeOf(p), "DMWithAttachments", func(_ *Plugin, _ string, _ ...*model.SlackAttachment) (string, error) {
+					return testutils.GetID(), nil
+				})
+
+				api.On("LogDebug", testutils.GetMockArgumentsWithType("string", 5)...).Return()
+			},
+			setupStore: func(s *mock_plugin.MockStore) {
+				s.EXPECT().StorePostIDs(testutils.GetID(), []string{testutils.GetID()}).Return(errors.New("unable to store post IDs"))
+			},
+		},
+		{
+			description: "error while sending attachments to the user and IsCharCountSafe returns false",
+			setupPlugin: func(p *Plugin, api *plugintest.API) {
+				monkey.PatchInstanceMethod(reflect.TypeOf(p), "DMWithAttachments", func(_ *Plugin, _ string, _ ...*model.SlackAttachment) (string, error) {
+					return testutils.GetID(), errors.New("error in sending attachments to the user")
+				})
+				monkey.PatchInstanceMethod(reflect.TypeOf(p), "IsCharCountSafe", func(_ *Plugin, _ []*model.SlackAttachment) bool {
+					return false
+				})
+			},
+			setupStore:    func(s *mock_plugin.MockStore) {},
+			expectedError: errors.New("error in sending attachments to the user"),
+		},
+	} {
+		t.Run(test.description, func(t *testing.T) {
+			defer monkey.UnpatchAll()
+			mockAPI := &plugintest.API{}
+			test.setupPlugin(&p, mockAPI)
+			p.SetAPI(mockAPI)
+			defer mockAPI.AssertExpectations(t)
+			mockCtrl := gomock.NewController(t)
+			mockedStore := mock_plugin.NewMockStore(mockCtrl)
+			test.setupStore(mockedStore)
+			p.store = mockedStore
+
+			err := p.HandleCarouselInput(testutils.GetID(), testutils.GetPickerBodyWithCarouselOptions(3))
+			if test.expectedError == nil {
+				require.Nil(t, err)
+			} else {
+				assert.EqualError(t, err, test.expectedError.Error())
+			}
+		})
+	}
+}
+
+func Test_HandlePreviousCarouselPosts(t *testing.T) {
+	p := Plugin{}
+	for _, test := range []struct {
+		description string
+		setupAPI    func(api *plugintest.API)
+		setupStore  func(s *mock_plugin.MockStore)
+	}{
+		{
+			description: "error in loading postIDs",
+			setupAPI: func(api *plugintest.API) {
+				api.On("LogDebug", testutils.GetMockArgumentsWithType("string", 5)...).Return()
+			},
+			setupStore: func(s *mock_plugin.MockStore) {
+				s.EXPECT().LoadPostIDs(testutils.GetID()).Return(nil, errors.New("error in loading post IDs"))
+			},
+		},
+		{
+			description: "no postIDs present in KV store",
+			setupAPI:    func(api *plugintest.API) {},
+			setupStore: func(s *mock_plugin.MockStore) {
+				s.EXPECT().LoadPostIDs(testutils.GetID()).Return([]string{}, nil)
+			},
+		},
+		{
+			description: "error in storing postIDs and getting post from API",
```

Comment:
```suggestion
			description: "error occurred while storing postIDs and getting post from API",
```
I think `while` sounds more clear than `in`, let me know your opinion